### PR TITLE
samtools mpileup in v1.9

### PR DESCRIPTION
**Pull request by bernt-matthias**
- [x] mpileup: 
  - [x]  upgrade to 1.8
  - [x] https://github.com/galaxyproject/tools-iuc/issues/1921
  - [x] compressed bcf output does not work yet: 
    - the latter two are obsolete in 1.9

### CODE REVIEW

File: tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml [hunk 1]
```diff
@@ -262,38 +262,57 @@ ${pasted_data}
...
+                    <param name="coefficient_for_modeling_homopolymer_errors" value="100" />
+                    <param name="skip_indel_calling_above_sample_depth" value="250" />
+                    <param name="gap_open_sequencing_error_probability" value="40" />
+                    <param name="platform_list_repeat" value="0" />
+                </conditional>
+            </conditional>
+            <conditional name="advanced_options">
+                <param name="advanced_options_selector" value="basic" />
+            </conditional>
+            <output name="output_file" file="samtools_mpileup_out_2.vcf" ftype="vcf" lines_diff="8" />
```
Comment (bgruening) [inline tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml]: This should not be output_file, but rather `output_file_vcf`

**Comment by bgruening** [comment]
I think your tests were just wrong, can this be?
Matthias, do you think we can put all the tools in one directory and use planemo demultiplexing to spit them put into different repos without the symlinking and keeping the currently repo working?

**Comment by bgruening** [comment]
> the output name was definitely wrong. odd that the planemo tests still worked. is this an issue?

We could be a lot smarter in linting and so on, but I guess no one of us has time to work on it.

> I'm wondering if the tool accepts all bam data types as input? Explicitely allowed is bam, but I guess also qname_sorted bam etc is accepted by the tool

Not sure. I would leave BAM, Galaxy will convert if needed and I assume an idex is required.

> for bam the index should be always present .. ?

Yes.

**Comment by nsoranzo** [comment]
> > the output name was definitely wrong. odd that the planemo tests still worked. is this an issue?

> We could be a lot smarter in linting and so on, but I guess no one of us has time to work on it.

Actually this is already checked by `planemo lint`, but it is only a warning, see the first Travis build for this PR: https://travis-ci.org/galaxyproject/tools-iuc/jobs/400921611

We should probably change it to an error.

**Comment by bernt-matthias** [comment]
made it an issue https://github.com/galaxyproject/planemo/issues/858

**Comment by mblue9** [comment]
@bernt-matthias did you see there's now a samtools v1.9 https://anaconda.org/bioconda/samtools

P.S Welcome to IUC!

**Comment by bernt-matthias** [comment]
@mblue9 thanks for the info. seems that they have deprecated samtools' mpileup in favour of bcftools. will see if I find the time.

**Comment by bernt-matthias** [comment]
Guess I'm done here: 
- updated to 1.9
- removed vcf and bcf output which is now deprected in `samtools mpileup` (they refer to bcftools)

**Comment by bernt-matthias** [comment]
One comment/question: since one of my samtools PR is already merged: how is this handled with the shared macros? Is each samtools tool deposited separately to the TS? Then I guess its OK, because each gets its separate copy of the macros.xml, or?

**Comment by mvdbeek** [comment]
Yeah, that should be ok, I don't think planemo detects this as a change, but we should not forget to adjust all to the latest macro version once we're done here.

**Comment by mvdbeek** [comment]
Thanks for all the much needed work, this is much appreciated!

**Pull request merged**